src/transmitter/proxy.ts

```diff
@@ -0,0 +1,56 @@
+import { URL } from 'url';
+import { httpOverHttp, httpsOverHttp } from 'tunnel';
+import { Agent, globalAgent as HttpAgent } from 'http';
+import { globalAgent as HttpsAgent } from 'https';
+
+import logger = require('../common/logger');
+
+export function getProxyAgent(
+  config: Record<string, any>,
+  endpoint: string,
+): Agent {
+  const url = new URL(endpoint);
+
+  // Check if the address is explicitly marked not to be proxied.
+  if (config.NO_PROXY) {
+    const hosts = config.NO_PROXY.split(',').map((host) => host.toLowerCase());
+
+    if (hosts.includes(url.hostname.toLowerCase())) {
+      return url.protocol === 'https:' ? HttpsAgent : HttpAgent;
+    }
+
+    // CIDR ranges are not supported, e.g. NO_PROXY=192.168.0.0/16.
+    // Wildcards are also not supported, e.g. NO_PROXY=*.mydomain.local
+    // HTTPS proxies (the connection to the proxy, not to the upstream) are not supported.
+  }
+
+  switch (url.protocol) {
+    case 'http:':
+      if (!config.HTTP_PROXY) {
```

Comment:
what does this mean? to return the httpagent if no HTTP_PROXY mentioned

Reply:
oh I got it :) never mind

Reply:
So Node.js has its own default HTTP/HTTPS agents which are used by networking libraries.
Here the function ensures to always return an agent. For http urls we return the default http agent; for https urls we return the default https agent.
Basically it means "use the default behaviour". The default agents are configured for normal, non-proxied requests